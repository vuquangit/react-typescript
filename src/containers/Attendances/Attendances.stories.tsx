import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import AppWrapper from '@/test/supports/AppWrapper'
import Attendances from '.'

export default {
  title: 'Story/Container/Attendances',
  component: Attendances,
  argTypes: {},
} as Meta

const TemplateLight: Story = (args) => (
  <AppWrapper themeMode="light">
    <Attendances {...args} />
  </AppWrapper>
)

export const Light = TemplateLight.bind({})

const TemplateDark: Story = (args) => (
  <AppWrapper themeMode="dark">
    <Attendances {...args} />
  </AppWrapper>
)

export const Dark = TemplateDark.bind({})
