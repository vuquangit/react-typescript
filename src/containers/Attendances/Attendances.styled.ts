import styled from 'styled-components'

export const AttendancesWrapper = styled('div')``
