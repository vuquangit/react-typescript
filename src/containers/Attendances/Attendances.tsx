import React, { FC } from 'react'
import CalendarWrappers from '@/components/CalendarWrapper'
import CalendarDetail from '@/components/AttendanceDetail'

const Attendances: FC = () => {
  return (
    <CalendarWrappers>
      <CalendarDetail />
    </CalendarWrappers>
  )
}

export default Attendances
