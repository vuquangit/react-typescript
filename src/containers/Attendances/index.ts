import Attendances from './Attendances'

export default Attendances
