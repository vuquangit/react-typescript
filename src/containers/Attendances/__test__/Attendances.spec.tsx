/**
 *
 * Tests for Calendar
 *
 */

import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import Attendances from '..'

import AppWrapper from '@/test/supports/AppWrapper'

let container: any = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  // cleanup on exiting
  document.body.removeChild(container)
  container = null
})

describe('App Calendar', () => {
  it('Test get default Calendar', async () => {
    // Test first render and componentDidMount
    await act(async () => {
      ReactDOM.render(
        <AppWrapper>
          <Attendances />
        </AppWrapper>,
        container
      )
    })

    expect(container).toMatchSnapshot()
  })
})
