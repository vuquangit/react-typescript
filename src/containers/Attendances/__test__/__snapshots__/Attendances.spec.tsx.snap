// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Calendar Test get default Calendar 1`] = `
.c3 {
  border: 0;
  font-size: 18px;
  cursor: pointer;
  border: 0;
  background-color: #000;
  color: #fff;
  font-size: 16px;
}

.c5 {
  padding: 8px;
  cursor: pointer;
  border: 0;
  background-color: #000;
  color: #fff;
  font-size: 16px;
}

.c10 {
  cursor: default;
  border: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border: 1px solid #dadada;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  padding: 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c7 {
  border-top: 0.05rem solid #e3e3e7;
  border-left: 0.05rem solid #e3e3e7;
  width: 100%;
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 7rem;
  border-bottom: 0.05rem solid #e3e3e7;
  border-right: 0.05rem solid #e3e3e7;
  cursor: default;
  pointer-events: none;
  max-width: 14.285714285714286%;
}

.c9:after {
  background: rgba(46,62,72,0.9);
  border-radius: 0.15rem;
  bottom: 100%;
  color: #fff;
  content: attr(data-tooltip);
  display: block;
  font-size: 0.8rem;
  left: 50%;
  max-width: 320px;
  opacity: 0;
  overflow: hidden;
  padding: 0.2rem 0.4rem;
  pointer-events: none;
  position: absolute;
  text-overflow: ellipsis;
  -webkit-transform: translate(-50%,0.4rem);
  -ms-transform: translate(-50%,0.4rem);
  transform: translate(-50%,0.4rem);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  white-space: pre;
  z-index: 300;
}

.c9:focus:after,
.c9:hover:after {
  opacity: 1;
  -webkit-transform: translate(-50%,-0.2rem);
  -ms-transform: translate(-50%,-0.2rem);
  transform: translate(-50%,-0.2rem);
}

.c12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 7rem;
  border-bottom: 0.05rem solid #e3e3e7;
  border-right: 0.05rem solid #e3e3e7;
  cursor: pointer;
  pointer-events: auto;
  max-width: 14.285714285714286%;
}

.c12:after {
  background: rgba(46,62,72,0.9);
  border-radius: 0.15rem;
  bottom: 100%;
  color: #fff;
  content: attr(data-tooltip);
  display: block;
  font-size: 0.8rem;
  left: 50%;
  max-width: 320px;
  opacity: 0;
  overflow: hidden;
  padding: 0.2rem 0.4rem;
  pointer-events: none;
  position: absolute;
  text-overflow: ellipsis;
  -webkit-transform: translate(-50%,0.4rem);
  -ms-transform: translate(-50%,0.4rem);
  transform: translate(-50%,0.4rem);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  white-space: pre;
  z-index: 300;
}

.c12:focus:after,
.c12:hover:after {
  opacity: 1;
  -webkit-transform: translate(-50%,-0.2rem);
  -ms-transform: translate(-50%,-0.2rem);
  transform: translate(-50%,-0.2rem);
}

.c11 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  height: 1.4rem;
  margin-right: 0.2rem;
  margin-top: 0.2rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0.05rem solid transparent;
  border-radius: 50%;
  color: #6c777e;
  cursor: pointer;
  font-size: 0.8rem;
  line-height: 1rem;
  outline: none;
  padding: 0.15rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 1.4rem;
  border-color: none;
  color: inherit;
}

.c15 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  height: 1.4rem;
  margin-right: 0.2rem;
  margin-top: 0.2rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0.05rem solid transparent;
  border-radius: 50%;
  color: #6c777e;
  cursor: pointer;
  font-size: 0.8rem;
  line-height: 1rem;
  outline: none;
  padding: 0.15rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 1.4rem;
  border-color: #cde4ff;
  color: #439aff;
}

.c13 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  line-height: 1;
  overflow-y: auto;
  padding: 0.2rem;
}

.c14 {
  border-radius: 0.15rem;
  font-size: 0.7rem;
  display: block;
  margin: 0.15rem auto;
  overflow: hidden;
  padding: 3px 4px;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #10b742;
  background: #d3f1dc;
}

@media screen and (min-width:40em) {
  .c3 {
    font-size: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
          font-size="18,24"
        >
          12/2020
        </button>
      </div>
      <div
        class="c4"
      >
        <button
          class="c5"
          cursor="pointer"
          data-testid="app-button"
        >
          &lt;&lt;
        </button>
        <button
          class="c5"
          cursor="pointer"
          data-testid="app-button"
        >
          &lt;
        </button>
        <button
          class="c5"
          cursor="pointer"
          data-testid="app-button"
        >
          Today
        </button>
        <button
          class="c5"
          cursor="pointer"
          data-testid="app-button"
        >
          &gt;
        </button>
        <button
          class="c5"
          cursor="pointer"
          data-testid="app-button"
        >
          &gt;&gt;
        </button>
      </div>
    </div>
    <div
      class="c6"
    >
      <div>
        Sun
      </div>
      <div>
        Mon
      </div>
      <div>
        Tue
      </div>
      <div>
        Wed
      </div>
      <div>
        Thu
      </div>
      <div>
        Fri
      </div>
      <div>
        Sat
      </div>
    </div>
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            29
          </button>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            30
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            1
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            2
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            3
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            4
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            5
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            6
          </button>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            7
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            8
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            9
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            10
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            11
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            12
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            13
          </button>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            14
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            15
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            16
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            17
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            18
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            19
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            20
          </button>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            21
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c15"
          >
            22
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            23
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            24
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            25
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            26
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            27
          </button>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            28
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            29
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            30
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            31
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c12"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c10 c11"
          >
            1
          </button>
          <div
            class="c13"
          >
            <span
              class="c14"
            >
              Attend OK
            </span>
            <span
              class="c14"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c9"
          data-tooltip="true"
        >
          <button
            class="c10 c11"
          >
            2
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
